src/java.base/share/classes/jdk/internal/constant/ConstantUtils.java

```diff
@@ -62,21 +66,21 @@ static String validateBinaryClassName(String name) {
     }
 
     /**
-      * Validates the correctness of an internal class name.
-      * In particular checks for the presence of invalid characters in the name.
-      *
-      * @param name the class name
-      * @return the class name passed if valid
-      * @throws IllegalArgumentException if the class name is invalid
-      */
-     static String validateInternalClassName(String name) {
-         for (int i=0; i<name.length(); i++) {
-             char ch = name.charAt(i);
-             if (ch == ';' || ch == '[' || ch == '.')
-                 throw new IllegalArgumentException("Invalid class name: " + name);
-         }
-         return name;
-     }
+     * Validates the correctness of an internal class name.
+     * In particular checks for the presence of invalid characters in the name.
+     *
+     * @param name the class name
+     * @return the class name passed if valid
+     * @throws IllegalArgumentException if the class name is invalid
+     */
+    public static String validateInternalClassName(String name) {
+        for (int i = 0; i < name.length(); i++) {
+            char ch = name.charAt(i);
+            if (ch == ';' || ch == '[' || ch == '.')
+                throw new IllegalArgumentException("Invalid class name: " + name);
```

Comment:
We can check there's no consecutive `..` `//` or trailing `.` or `/` so we can just use the validated string to create a reference class desc.

Reply:
Sounds reasonable, but I think I have already piled on too much into this PR. Ok to defer to a follow-up?